### ISSUE TITLE
Tests fail after finish when using Flutter 3.10

### ISSUE THREAD

**Issue by bartekpacia** [leancodepl/patrol#1265]
When running Patrol tests on Flutter 3.10, the tests start and run just fine, but when they finish, they fail with

```
java.lang.Exception: ══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞═════════════════
The following assertion was thrown running a test:
A SemanticsHandle was active at the end of the test.
All SemanticsHandle instances must be disposed by calling
dispose() on the SemanticsHandle.
When the exception was thrown, this was the stack:
#0 WidgetTester._verifySemanticsHandlesWereDisposed (package:flutter_test/src/widget_tester.dart:1051:7)
#1 WidgetTester._endOfTestVerifications (package:flutter_test/src/widget_tester.dart:1042:5)
#2 TestWidgetsFlutterBinding._runTestBody (package:flutter_test/src/binding.dart:998:22)
<asynchronous suspension>
The test description was:
counter state is the same after going to Home and switching
apps
═════════════════════════════════════════════════════════════════
at pl.leancode.patrol.PatrolTestRunner.run(PatrolTestRunner.kt:76)
at org.junit.runners.Suite.runChild(Suite.java:128)
at org.junit.runners.Suite.runChild(Suite.java:27)
at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
at androidx.test.internal.runner.TestExecutor.execute(TestExecutor.java:67)
at androidx.test.internal.runner.TestExecutor.execute(TestExecutor.java:58)
at androidx.test.runner.AndroidJUnitRunner.onStart(AndroidJUnitRunner.java:446)
at android.app.Instrumentation$InstrumentationThread.run(Instrumentation.java:2361)
```

It didn't occur on my emulator and physical device when I was testing how Patrol works on `3.10.0-1.5.pre` yesterday.

**Comment by bartekpacia** [comment]
The first Google search result: https://github.com/flutter/flutter/issues/18015

**Comment by mashedbymachines** [comment]
We have the same issue on an Android 12 device and an Android 11 device in our CI. I dont get the error when running on my personal Android 13 device though.

I/flutter (20496): ══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
I/flutter (20496): The following assertion was thrown running a test:
I/flutter (20496): A SemanticsHandle was active at the end of the test.
I/flutter (20496): All SemanticsHandle instances must be disposed by calling dispose() on the SemanticsHandle.
I/flutter (20496): 
I/flutter (20496): When the exception was thrown, this was the stack:
I/flutter (20496): #0      WidgetTester._verifySemanticsHandlesWereDisposed (package:flutter_test/src/widget_tester.dart:1051:7)
I/flutter (20496): #1      WidgetTester._endOfTestVerifications (package:flutter_test/src/widget_tester.dart:1042:5)
I/flutter (20496): #2      TestWidgetsFlutterBinding._runTestBody (package:flutter_test/src/binding.dart:998:22)
I/flutter (20496): <asynchronous suspension>

**Comment by bartekpacia** [comment]
Hi @mashedbymachines, thanks for chipping in. I understand you're experiencing this problem on Flutter 3.10, and it worked fine on previous versions?

**Comment by mashedbymachines** [comment]
> 
Yes everything worked fine on flutter 3.7.12.

The problem appeared after we updated to flutter 3.10 from 3.7.12 on our CI machine yesterday and suddenly we get this problem when running patrol tests. (Patrol CLI v1.1.9) (Both tested with patrol pub package 1.1.0 and 1.1.4)

The problem occurs on our master branch where flutter 3.7.12 is used, so the project didnt even need to update to use a newer flutter version to get the issue.

**Comment by mashedbymachines** [comment]
Update:
We downgraded flutter to 3.7.12 in our CI system and the tests started working again.

**Comment by bartekpacia** [comment]
Nice. That's what we're doing as well for now, until we find a fix.

**Comment by bartekpacia** [comment]
I'm pretty certain this PR is a source of this bug: https://github.com/flutter/flutter/pull/121571

Another candidate: https://github.com/flutter/flutter/pull/121289

Related files:
- https://github.com/flutter/flutter/commits/master/packages/integration_test/lib/integration_test.dart
- https://github.com/flutter/flutter/commits/master/packages/flutter_test/lib/src/binding.dart

**Comment by sonnyvesali** [comment]
@bartekpacia I'm getting an issue on iOS saying 

```[!] CocoaPods could not find compatible versions for pod "patrol":
  In Podfile:
    patrol (from `.symlinks/plugins/patrol/ios`)

Specs satisfying the `patrol (from `.symlinks/plugins/patrol/ios`)` dependency were found, but they required a higher minimum deployment target.```

Possibly related? 

I remember I got some grief from XCode when I tried to build the app after upgrading to 3.10.0, but is this resolved, or should I just downgrade

**Comment by Kendru98** [comment]
For me, patrol with flutter 3.10 working as before. 
```
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel stable, 3.10.0, on Microsoft Windows [Version 10.0.19045.2604], locale pl-PL)
[√] Windows Version (Installed version of Windows is version 10 or higher)
[√] Android toolchain - develop for Android devices (Android SDK version 33.0.0-rc2)
[√] Chrome - develop for the web
[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.11.11)
[√] Android Studio (version 2021.1)
[√] IntelliJ IDEA Community Edition (version 2021.1)
[√] VS Code (version 1.78.2)
[√] Connected device (4 available)
[√] Network resources
```

**Comment by ziadsarour** [comment]
I have the same issue, I don't know how if it worked on previous versions as I'm using Patrol since Flutter > 3.10

<details>
<summary>flutter doctor --verbose</summary>
<br>
[✓] Flutter (Channel stable, 3.10.4, on macOS 13.4 22F66
    darwin-x64, locale fr-FR)
    • Flutter version 3.10.4 on channel stable at
      /usr/local/Caskroom/flutter/3.10.4/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 682aa387cf (7 days ago), 2023-06-05
      18:04:56 -0500
    • Engine revision 2a3401c9bb
    • Dart version 3.0.3
    • DevTools version 2.23.1

[✓] Android toolchain - develop for Android devices (Android SDK
    version 31.0.0)
    • Android SDK at /usr/local/share/android-sdk
    • Platform android-34, build-tools 31.0.0
    • ANDROID_HOME = /usr/local/share/android-sdk
    • ANDROID_SDK_ROOT = /usr/local/share/android-sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E300c
    • CocoaPods version 1.12.1

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android
      Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.79.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.66.0

[✓] Connected device (1 available)
    • sdk gphone x86 64 (mobile) • emulator-5554 • android-x64 •
      Android 11 (API 30) (emulator)

[✓] Network resources
    • All expected network resources are available.

• No issues found!

</details>

**Comment by bartekpacia** [comment]
Workaround has been implemented in #1352 and released in [`patrol` v1.1.5](https://pub.dev/packages/patrol/changelog#115)
